src/storable.rs

```diff
@@ -492,6 +492,102 @@ where
     };
 }
 
+fn serialize_bounded_with_size<T>(entry: &T, bytes: &mut [u8], start: usize) -> usize
+where
+    T: Storable,
+{
+    let bounds = bounds::<T>();
+    let size_len = bytes_to_store_size(&bounds) as usize;
+    let max_size = bounds.max_size as usize;
+    let entry_bytes = entry.to_bytes();
+    debug_assert!(entry_bytes.len() <= max_size);
+
+    encode_size(
+        &mut bytes[start..start + size_len],
+        entry_bytes.len(),
+        &bounds,
+    );
+
+    bytes[start + size_len..start + size_len + entry_bytes.len()]
+        .copy_from_slice(entry_bytes.borrow());
+
+    start + max_size + size_len
+}
+
+fn deserialize_bounded_with_size<T>(bytes: &[u8], start: usize) -> (T, usize)
+where
+    T: Storable,
+{
+    let bounds = bounds::<T>();
+    let max_size = bounds.max_size as usize;
+    let size_len = bytes_to_store_size(&bounds) as usize;
+    let len = decode_size(&bytes[start..start + size_len], &bounds);
+
+    let a = T::from_bytes(Cow::Borrowed(
+        &bytes[start + size_len..start + size_len + len],
+    ));
+    (a, start + max_size + size_len)
+}
+
+impl<A, B, C> Storable for (A, B, C)
+where
+    A: Storable,
+    B: Storable,
+    C: Storable,
+{
+    fn to_bytes(&self) -> Cow<[u8]> {
```

Comment:
Would it not make sense to re-use code between this implementation and the `(A, B)` implementation?

Reply:
I think it's possible, but per my other comment, I think the current implementation of `(A, B)` is inefficient, as we're needlessly storing the sizes when elements are fixed, so better keep this implementation separate so that we can optimize it better.